### DISCUSSION TITLE
Authorizer-interface changed in 3.0, what is javax.ws.rs.container.ContainerRequestContext about?

### DISCUSSION THREAD

**Discussion by pvdh** [dropwizard/dropwizard#7871] [Q&A]
In the past I've been using an authorizer according to https://www.dropwizard.io/en/stable/manual/auth.html#authorizer, the interface was `public boolean authorize(User user, String role)`.

It has been changed to include a ContainerRequestContect:

`public boolean authorize(User user, String role, ContainerRequestContext ctx)`

What is this Context about? Can I safely ignore it?

Best regards
Patrick

**Comment by joschi** [comment]
See https://github.com/dropwizard/dropwizard/pull/2837 for the related change (which was introduced in Dropwizard 2.0.0 already).

`ctx` is nullable and you can ignore it in your implementation if you're not interested in getting any information about the currently handled JAX-RS/Jersey request for which the authorization should be checked.

See also https://javaee.github.io/javaee-spec/javadocs/javax/ws/rs/container/ContainerRequestContext.html.
